File: solutions/swift/resistor-color/1/Sources/ResistorColor/ResistorColor.swift
enum ResistorColorError: Error {
    case invalidColorName
}

enum ResistorColor: String, CaseIterable {
    case black, brown, red, orange, yellow, green, blue, violet, grey, white

    static let colors = allCases.map(\.rawValue)

    static func colorCode(for name: String) throws -> Int {
        guard let color = ResistorColor(rawValue: name),
              let index = allCases.firstIndex(of: color) else {
            throw ResistorColorError.invalidColorName
        }

        return index
    }
}

File: solutions/vbnet/resistor-color/1/ResistorColor.vb
Module Program
    Function ColorCode(ByVal color As String) As Integer
        Return Array.IndexOf(Colors(), color)
    End Function

    Function Colors() As String()
        Return {"black", "brown", "red", "orange", "yellow", _
                "green", "blue", "violet", "grey", "white"}
    End Function
End Module